### PR TITLE
MEGA-516: Add script to convert dylibs to frameworks

### PR DESCRIPTION
**Pull request by jstys92**
Adds a script which takes the thin dylibs and converts each of them into an iOS framework. The existing framework building script bundled all the static libs into a single mega target, and in the end was still a static link - so not quite what we wanted.

I was able to run with the frameworks this builds, so other than any additional nitpicks from AppStore connect that I'm still checking, it should be good for review.

### CODE REVIEW

File: build_ios/README.md [hunk 1]
```diff
@@ -33,6 +33,10 @@ Use build-ffmpeg-tvos.sh for tvOS.
 * To build fat libraries from separately built thin libraries:
 
         ./build-ffmpeg.sh lipo
```
Comment (DillonSkeehan) [inline build_ios/README.md:35]: okay I didn't catch this before but to call the fat libraries lipo is hilarious.

**Review by DillonSkeehan — APPROVED** [review]

**Comment by DillonSkeehan** [comment]
So if I understand this correctly, we can now dynamic link to each of the needed ffmpeg libraries?

**Comment by jstys92** [comment]
We could do that already directly with the dylibs, but they didn't have the neat framework packaging around them, so would get rejected by the AppStore. This whole change was needed so essentially the same files can be inside framework folders, have a .plist file added and then have their rpaths updated so they still find each other.

I only mentioned the other framework building script to address that something like this was here already, but it wasn't quite what we wanted

**Comment by DillonSkeehan** [comment]
ah isee

**Pull request merged**